### PR TITLE
Allow for Cortex Search instead of just Pinecone.

### PR DESCRIPTION
**Pull request by sfc-gh-dkurokawa**
Note we have to use python 3.11 since the Snowflake python packages aren't available for 3.12 yet.

### CODE REVIEW

File: trulens_eval/examples/snowflake_demo/retrieve.py [hunk 1]
```diff
@@ -36,3 +36,38 @@ def retrieve(self, query: str):
...
+        connection_parameters = {
+            "account": os.environ["SF_ACCOUNT"],
+            "user": os.environ["SF_USER"],
+            "password": os.environ["SF_PASSWORD"],
+            "role": os.environ["SF_ROLE"],
+            "warehouse": os.environ["SF_WAREHOUSE"],
+        }
+        session = Session.builder.configs(connection_parameters).create()
+        root = Root(session)
+        self._cortex_search_service = root.databases[
```
Comment (sfc-gh-dhuang) [inline trulens_eval/examples/snowflake_demo/retrieve.py:54]: what's the reason for using `Root` here to traverse the SF databases if we are only using the single specified DB and SCHEMA?

couldn't we just pass `SF_DB_NAME` and `SF_SCHEMA` into the connection_parameters?

File: trulens_eval/examples/snowflake_demo/requirements.txt [hunk 2]
```diff
@@ -14,6 +14,7 @@ llama-index-embeddings-huggingface==0.2.0
 llama-index-legacy==0.9.48
 litellm==1.38.12
 streamlit==1.35
+snowflake[ml]
```
Comment (sfc-gh-chu) [inline trulens_eval/examples/snowflake_demo/requirements.txt:17]: should this be `snowflake-ml-python`?

File: trulens_eval/examples/snowflake_demo/retrieve.py [hunk 3]
```diff
@@ -36,3 +36,38 @@ def retrieve(self, query: str):
...
+            os.environ["SF_DB_NAME"]].schemas[
+                os.environ["SF_SCHEMA"]].cortex_search_services[
+                    os.environ["SF_CORTEX_SEARCH_SERVICE"]]
+
+    def retrieve(self, query: str):
```
Comment (sfc-gh-jreini) [inline trulens_eval/examples/snowflake_demo/retrieve.py:59]: Can you match the pinecone retriever's use of the context filter (or remove from both)?

File: trulens_eval/examples/snowflake_demo/requirements.txt (hunk 2 above)
Comment (sfc-gh-dkurokawa) [reply trulens_eval/examples/snowflake_demo/requirements.txt:17]: I forget why, but at the time that didn't work and the docs for cortex search:
https://docs.snowflake.com/LIMITEDACCESS/cortex-search/query-cortex-search-service#label-cortex-search-query-syntax-python
Further linked to this:
https://docs.snowflake.com/developer-guide/snowflake-python-api/snowflake-python-installing
Which said to do `pip install snowflake[ml] -U` at the bottom.

File: trulens_eval/examples/snowflake_demo/retrieve.py (hunk 1 above)
Comment (sfc-gh-dkurokawa) [reply trulens_eval/examples/snowflake_demo/retrieve.py:54]: oh I was just following what the cortex search doc said to do:
https://docs.snowflake.com/LIMITEDACCESS/cortex-search/query-cortex-search-service#label-cortex-search-query-syntax-python
TBH, the service is kinda brittle so I figured I should just follow what they said exactly haha.

File: trulens_eval/examples/snowflake_demo/retrieve.py (hunk 3 above)
Comment (sfc-gh-dkurokawa) [reply trulens_eval/examples/snowflake_demo/retrieve.py:59]: sorry, what do you mean?

**Review by sfc-gh-chu — APPROVED** [review]

File: trulens_eval/examples/snowflake_demo/retrieve.py (hunk 3 above)
Comment (sfc-gh-jreini) [reply trulens_eval/examples/snowflake_demo/retrieve.py:59]: I meant that the PineconeRetriever uses `WithFeedbackFilterDocuments` (although with filter set to 0), while cortex search doesn't here. Addressing my comment here: https://github.com/truera/trulens/pull/1229

**Review by sfc-gh-jreini — APPROVED** [review]

**Pull request merged**